build-support/bin/black.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+cd "${REPO_ROOT}" || exit 1
+
+# shellcheck source=build-support/common.sh
+source build-support/common.sh
+
+function usage() {
+  echo "Formats python files with black through pants"
+  echo
+  echo "Usage: $0 (-h|-f)"
+  echo " -h    print out this help message"
+  echo " -f    instead of erroring on files with incorrect formatting, fix"
+  echo "       those files"
+  if (( $# > 0 )); then
+    die "$@"
+  else
+    exit 0
+  fi
+}
+
+black_args=(
+  --check
+)
+
+while getopts "hf" opt
+do
+  case ${opt} in
+    h) usage ;;
+    f) black_args=() ;;
+    *) usage "Invalid option: -${OPTARG}" ;;
+  esac
+done
+
+./pants --changed-parent="$(git_merge_base)" fmt-v2 --black-options="${black_args[*]}"
```

Comment:
Encouraged to write this in Python because you can then use argparse for short and long options (`-f` and `--fix`) + general code benefits like type hints.

Example of how to run the subprocess and propagate the error code: https://github.com/pantsbuild/pants/blob/master/build-support/bin/shellcheck.py

Example of how to set up argparse: 
https://github.com/pantsbuild/pants/blob/77caa911164e3c8057c43405042e0ef1995382d1/build-support/bin/check_header.py#L59-L77

https://github.com/pantsbuild/pants/blob/77caa911164e3c8057c43405042e0ef1995382d1/build-support/bin/check_header.py#L40

Reply:
Done and extracted into its own PR: https://github.com/pantsbuild/pants/pull/8454 (hook will come later after reformatting)